processing/src/main/java/org/apache/druid/segment/Metadata.java

```diff
@@ -233,6 +260,39 @@ public String toString()
            ", timestampSpec=" + timestampSpec +
            ", queryGranularity=" + queryGranularity +
            ", rollup=" + rollup +
+           ", sortOrder=" + sortOrder +
            '}';
   }
+
+  @Nullable
+  private static List<String> mergeSortOrders(List<List<String>> sortOrdersToMerge)
+  {
+    if (sortOrdersToMerge.isEmpty()) {
+      return null;
+    }
+
+    if (sortOrdersToMerge.stream().anyMatch(Objects::isNull)) {
+      return null;
+    }
+
+    final List<String> mergedSortOrder = new ArrayList<>();
+
+    while (true) {
+      String column = null;
+
+      for (final List<String> sortOrder : sortOrdersToMerge) {
+        if (mergedSortOrder.size() >= sortOrder.size()) {
```

Comment:
does this happen from columns with all null values or something? maybe this method could use some comments to make the rationale behind the decisions clearer

Reply:
I added some comments and also fixed a problem: `null` is now treated as `[__time]` (which makes sense, since that was the only possible sort order prior to the `sortOrder` field being added). Due to this change the method is no longer `@Nullable`.